src/ChipInput.js

```diff
@@ -250,9 +256,19 @@ class ChipInput extends React.Component {
   handleKeyDown = (event) => {
     const { focusedChip } = this.state
     this.setState({ keyPressed: false, preventChipCreation: false })
+    if (this.props.onKeyDown) {
+      // Needed for arrow controls on menu in autocomplete scenario
+      this.props.onKeyDown(event)
+      // Check if the callback marked the event as isDefaultPrevented() and skip further actions
+      // enter key for example should not always add the current value of the inputField
+      if (event.isDefaultPrevented()) {
+        return
+      }
+    }
 
     if (this.props.newChipKeyCodes.indexOf(event.keyCode) >= 0) {
-      this.handleAddChip(event.target.value)
+      let result = this.handleAddChip(event.target.value)
+      if (result !== false) event.preventDefault()
```

Comment:
We should document what the return value of `handleAddChip` means and ensure that it _always_ returns a boolean (instead of `false` and `undefined`).